missingNumbers.js

```diff
@@ -0,0 +1,25 @@
+function findMissingNumbers(inputArray) {
+
+    var missingNumbers = [];
+    if (typeof inputArray === 'string') {
+        return 'invalid input'
+    }
+    if (inputArray.length <= 1) {
+        return 'array length too small'
+    }
+    inputArray = inputArray.sort()
+    var i = 1
+    var missingValue = inputArray[0] + 1;
+    const count = ((inputArray[inputArray.length - 1]) - missingValue) - (inputArray.length - 2);
+    while (missingNumbers.length < count) {
+        while (inputArray[i] !== missingValue) {
+            missingNumbers.push(missingValue++)
+        }
+        i++;
+        missingValue++;
+    }
+    return missingNumbers;
+}
+/*var ar = [ 4,9,1 ];
```

Comment:
It's always a good practice to remove commented code. It's always a sign that you don't need it.
Comments are usually used to explain why you are doing something in code